cats/cats-dynomite/src/main/java/com/netflix/spinnaker/cats/dynomite/cluster/DynoClusteredAgentScheduler.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.cats.dynomite.cluster;
+
+import com.netflix.spinnaker.cats.agent.Agent;
+import com.netflix.spinnaker.cats.agent.AgentExecution;
+import com.netflix.spinnaker.cats.agent.AgentLock;
+import com.netflix.spinnaker.cats.agent.AgentScheduler;
+import com.netflix.spinnaker.cats.agent.AgentSchedulerAware;
+import com.netflix.spinnaker.cats.agent.ExecutionInstrumentation;
+import com.netflix.spinnaker.cats.dynomite.DynomiteClientDelegate;
+import com.netflix.spinnaker.cats.module.CatsModuleAware;
+import com.netflix.spinnaker.cats.redis.cluster.AgentIntervalProvider;
+import com.netflix.spinnaker.cats.redis.cluster.ClusteredAgentScheduler;
+import com.netflix.spinnaker.cats.redis.cluster.NodeIdentity;
+import com.netflix.spinnaker.cats.redis.cluster.NodeStatusProvider;
+import com.netflix.spinnaker.cats.thread.NamedThreadFactory;
+import redis.clients.jedis.JedisCommands;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Temporary clustered agent scheduler while we're waiting for Dyno client support of evalsha and loadscript.
+ *
+ * Shares a similiar strategy as ClusteredAgentScheduler, but doesn't use Lua, is slower and unsafe (does not use atomic
```

Comment:
This is actually a bit dated, now that we're doing `setnx` we should be a lot safer. This is our deployment-specific, but since we're running consistency of `ONE` in dyno_clouddriver, there is a chance for running multiple times if things are just right.